lib/rainforest/cli/test_importer.rb

```diff
@@ -312,4 +292,49 @@ def create_test_obj(rfml_test)
 
     test_obj
   end
+
+  def make_test_priority_groups
+    # Prioritize embedded tests before other tests
+    upload_groups = []
+    unordered_tests = []
+    queued_tests = test_files.test_data.dup
+
+    until queued_tests.empty?
+      new_ordered_group = []
+      ordered_ids = upload_groups.flatten.map(&:rfml_id)
+
+      queued_tests.each do |rfml_test|
+        if (rfml_test.embedded_ids - ordered_ids).empty?
+          new_ordered_group << rfml_test
+        else
+          unordered_tests << rfml_test
+        end
+      end
+
+      # If all the queued tests make it to the unordered tests group, then
+      # they contain non-existent RFML ids.
```

Comment:
This isn't obvious to me at all. Can you either explain it better of figure out a more obvious test?

In general, I feel like this whole method would be much clearer if it were recursive--I can't really keep track of what all the different groups are, and they seem to be kind of oddly named. What does this method do exactly?